rfcs/text/0009_screenshot_mode_service.md

```diff
@@ -0,0 +1,151 @@
+- Start Date: 2020-03-02
+- RFC PR: (leave this empty)
+- Kibana Issue: (leave this empty)
+
+# Summary
+
+Currently, the applications that support screenshot reports are:
+ - Dashboard
+ - Visualize Editor
+ - Canvas
+
+Kibana UI code should be aware when the page is rendering for the purpose of
```

Comment:
I wonder if screenshot mode ownership and performance problems are essentially the same problems that we have and will have to address at some point for dashboard embed mode. 

* Loading a single dashboard in embed mode loads too much code
* There are a lot of interactivity bugs. Most of them due to the only dashboard is aware of embed mode, but not lower-level plugins https://github.com/elastic/kibana/issues/93200


So I am genuinely curious if screenshot mode should be generalized into `static` mode. Handle all screenshots, embeds, and lighter view-only kibana use cases. Become part of the core to be able to optimize on-page load and on plugin bundling/loading level.

Reply:
there are similar but different requirements i think. the most notable is that in embedded mode you most likely still want to collect telemetry and probably you still want a bit more interactivity that you would in report.